test/unit/remote-config/remote-config.spec.ts

```diff
@@ -0,0 +1,231 @@
+    });
+
+    it('should reject when API response does not contain an ETag', () => {
+      const response = deepCopy(REMOTE_CONFIG_RESPONSE);
+      response.eTag = '';
+      const stub = sinon
+        .stub(RemoteConfigApiClient.prototype, 'getTemplate')
+        .resolves(response);
+      stubs.push(stub);
+      return remoteConfig.getTemplate()
+        .should.eventually.be.rejected.and.have.property(
+          'message', `Invalid Remote Config template response: ${JSON.stringify(response)}`);
+    });
+
+    it('should reject when API response does not contain valid parameters', () => {
+      const response = deepCopy(REMOTE_CONFIG_RESPONSE);
+      response.parameters = null;
+      const stub = sinon
+        .stub(RemoteConfigApiClient.prototype, 'getTemplate')
+        .resolves(response);
+      stubs.push(stub);
+      return remoteConfig.getTemplate()
+        .should.eventually.be.rejected.and.have.property(
+          'message', `Remote Config parameters must be a non-null object`);
+    });
+
+    it('should reject when API response does not contain valid conditions', () => {
+      const response = deepCopy(REMOTE_CONFIG_RESPONSE);
+      response.conditions = Object();
+      const stub = sinon
+        .stub(RemoteConfigApiClient.prototype, 'getTemplate')
+        .resolves(response);
+      stubs.push(stub);
+      return remoteConfig.getTemplate()
+        .should.eventually.be.rejected.and.have.property(
+          'message', `Remote Config conditions must be an array`);
+    });
+
+    it('should resolve with Remote Config template on success', () => {
+      const stub = sinon
+        .stub(RemoteConfigApiClient.prototype, 'getTemplate')
+        .resolves(REMOTE_CONFIG_RESPONSE);
+      stubs.push(stub);
+
+      return remoteConfig.getTemplate()
+        .then((template) => {
+          expect(template.conditions.length).to.equal(1);
+          expect(template.conditions[0].name).to.equal('ios');
+          expect(template.conditions[0].expression).to.equal('device.os == \'ios\'');
+          // verify the property mapping in RemoteConfigCondition from `tagColor` to `color`
+          expect(template.conditions[0].color).to.equal('BLUE');
+
+          expect(template.eTag).to.equal('etag-123456789012-5');
+
+          const key = 'holiday_promo_enabled';
+          const p1 = template.parameters[key];
+          expect(p1.defaultValue).deep.equals({ value: 'true' });
+          expect(p1.conditionalValues).deep.equals({ ios: { useInAppDefault: true } });
+          expect(p1.description).equals('this is a promo');
+
+          const p2 = template.getParameter(key);
+          if (p2) {
+            expect(p2.defaultValue).deep.equals({ value: 'true' });
+            expect(p2.conditionalValues).deep.equals({ ios: { useInAppDefault: true } });
+            expect(p2.description).equals('this is a promo');
+          }
+
+          const c = template.getCondition('ios');
+          if (c) {
```

Comment:
`expect(c).to.be.not.undefined`

Reply:
It looks like even with the above assert, without the `if` I get the following error `remote-config.spec.ts (211,18): Object is possibly 'undefined'.`

Also, the docs recommend that it is best to asset the target is equal to its expected value, instead to the negation. https://www.chaijs.com/api/bdd/#method_undefined

Should we keep both?

Reply:
```
expect(c).to.be.not.undefined;
const cond = c as RemoteConfigCondition;
// other assertions on cond
```